docs/reference/configuration/configuration.md

```diff
@@ -1,48 +1,276 @@
-## Configuration overview
+## The configuration system
 
-The Arm Mbed configuration system customizes the compile time configuration of Mbed components: targets, libraries and applications.
+The Arm Mbed OS configuration system, a part of the Arm Mbed OS build tools, customizes compile time configuration parameters. Each library may define a number of configuration parameters in its `mbed_lib.json`. `mbed_app.json` may override the values of these configuration parameters. Configuration is defined using [JSON](http://www.json.org/). Some examples of configuration parameters:
```

Comment:
Query: Who or what defines configuration using JSON? You, the user? Please clarify for active voice.

Reply:
The sentence above outlines where confutation comes from: libraries, targets, applications.